src/random.hpp

```diff
@@ -2,15 +2,66 @@
 
 #include <cstdint>
 #include <cstdio>
+#if defined(__linux__) || defined(__APPLE__)
+#elif defined(_WIN32)
+#include <windows.h>
+#include <bcrypt.h>
+#pragma comment( lib, "Bcrypt" )
+#else
+#error "Unsupported OS"
+#endif
+
 
 class Random
 {
+
+#if defined(__linux__) || defined(__APPLE__)
+public:
+	Random() {
+		file = fopen("/dev/urandom", "rb");
+	}
+	~Random() {
+		if (file)
+			fclose(file);
+	}
+
+	bool get(void* data, std::size_t size) {
+		fread(data, size, 1, file);
+		return true;
```

Comment:
```suggestion
		return fread(data, size, 1, file) == size;
```
Check for in in `fread()`.